contribs/drt/src/main/java/org/matsim/contrib/drt/stops/StopDurationProvider.java

```diff
@@ -0,0 +1,15 @@
+package org.matsim.contrib.drt.stops;
+
+import org.matsim.contrib.drt.passenger.DrtRequest;
+import org.matsim.contrib.dvrp.fleet.DvrpVehicle;
+
+/**
+ * This class provides the stop duration of a request. Eventually, this could
+ * also become a simple attribute of a DrtRequest to simplify the code
+ * complexity.
+ */
+public interface StopDurationProvider {
```

Comment:
If we assume the stop duration is a linear function (a fixed duration + sum of stop duration of each request), we could call it `StopDurationIncreaseCalculator` or `PassengerStopDurationCalculator` or, following your suggestion, `StopInteractionDurationCalculator`. And accordingly change the method names. Perhaps the last one is best if we want to use some non-linear functions in the future.

I would rather not put it into the request (as a standard attribute), because it is very hard to later remove/modify it.

Reply:
So taking the suggestions, I would propose `PassengerStopDurationProvider`. I would like to call it "provider" to convey that we really just want to obtain a value here, the arbitrarily complex calculation should happen in the specific `StopTimeCalculator`.